woocommerce/woocommerce
# Add editable headings for cart and checkout 'Total' heading

### Changes proposed in this Pull Request:

Partially addresses [#52043](https://github.com/woocommerce/woocommerce/issues/52043).

In https://github.com/woocommerce/woocommerce/pull/52892 I implemented editable headings via entities, but overall didn't like the experience or the needless added complexity.

This takes a more traditional route with block attributes and the experience feels much better. Also the last PR was much too large so I've opted to break it into smaller PR's to address each heading one at a time. Easier for reviewing and testing.

This allows the editing of the order summary "Total" heading in both the cart and checkout blocks. 



### How to test the changes in this Pull Request:

For cart and checkout:

* Edit the page, find the "Total" in the order summary, click on the text to edit it
* Smoke test various editing scenarios (empty out all the text, save empty, etc)
* Save the changes
* Smoke test the changes display as expected on the frontend.

Test translations:

* Have translations installed for Nederlands (but stay in English for now)
* Edit the page find the "Total" heading, edit it to anything and save
* Now go back and edit the page again setting the heading to the english default Subtotal
* Now swap to Nederlands language in WordPress
* Now go to front-end and see that the heading should be translated to Totaal



### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Allow merchants to edit the cart and checkout block order summary total headings.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
